### ISSUE TITLE
Keras contrib module

### ISSUE THREAD

**Issue by bbengfort** [DistrictDataLabs/yellowbrick#1106]
**Describe the solution you'd like**

Create a contrib module, `yellowbrick.contrib.keras` that better supports the use of Keras visualizers with Yellowbrick and creates the opportunity to create Keras-specific (or neural network-specific) visualizers. The contrib model would implement or extend the `KerasClassifier` and `KerasRegressor` estimators and also provide a functional API for using keras models in Yellowbrick. 

- [ ] Create `KerasClassifier` and `KerasRegressor` wrappers in `yellowbrick.contrib.keras`
- [ ] Test the use of Keras with the most popular Yellowbrick Visualizers 
- [ ] Create issues for Keras-specific visualizers that might be useful for Yellowbrick 

**Is your feature request related to a problem? Please describe.**

This PR is related to #397 that described using Yellowbrick with other ML libraries. Since this discussion, Yellowbrick has incorporated contrib support for 3rd party libraries using wrappers and other methods (see #1103) and has been [used successfully with Keras](https://towardsdatascience.com/evaluating-keras-neural-network-performance-using-yellowbrick-visualizations-ad65543f3174) using the `keras.wrappers.scikit_learn.KerasClassifier` wrapper. However, from that same blog post:

> Unfortunately, if our goal is to use Yellowbrick visualizations on Keras models, KerasClassifier/KerasRegressor doesn’t quite go far enough. 😞 There are a couple of problems:
>
> 1. Yellowbrick relies on some Scikit-Learn model semantics that KerasClassifier doesn’t provide.
> 1. KerasClassifier doesn’t provide alternative training methods, like fit_generator or fit_dataframe, which are hard to give up.

The blog post (by the way, thanks again @ResidentMario!) describes a `KerasBatchClassifier` implementation that resolves some of these issues, but making this a first class `contrib` module would allow more advanced usage for things that aren't as easily emulated. 

**Examples**

```python
import numpy as np
from keras.wrappers.scikit_learn import KerasClassifier
from sklearn.base import BaseEstimator

class KerasBatchClassifier(KerasClassifier, BaseEstimator):
    def __init__(self, model, **kwargs):
        super().__init__(model)
        self.fit_kwargs = kwargs
        self._estimator_type = 'classifier'

    def fit(self, *args, **kwargs):
        # taken from keras.wrappers.scikit_learn.KerasClassifier.fit
        self.model = self.build_fn(**self.filter_sk_params(self.build_fn))
        
        self.classes_ = np.array(range(len(self.fit_kwargs['train_generator'].class_indices)))
        self.__history = self.model.fit_generator(
            self.fit_kwargs.pop('train_generator'), 
            **self.fit_kwargs
        )
```

![Keras Model Loss Plot](https://miro.medium.com/max/700/1*xI_O3NMCTIhEXHZ6Dl92ng.png)

@mattharrison @carlomazzaferro @ndanielsen @ResidentMario any other comments on this?

@carlomazzaferro should we create a separate issue for [skorch](https://github.com/skorch-dev/skorch) or does Skorch work out of the box?

**Comment by ResidentMario** [comment]
Very cool!
